### PR TITLE
Feature/tri 290 swagger documentation irs after merge main

### PR DESCRIPTION
**Pull request by ds-jkreutzfeld**

### CODE REVIEW

File: irs-models/src/main/java/net/catenax/irs/component/Job.java [hunk 1]
```diff
@@ -60,38 +62,58 @@ public class Job {
...
+    @Schema(description = "HTTP verbs used by request.")
     private String action;
 
     /**
      * Owner of the job
      */
+    @Schema(description = "The requestor of the request.")
     private String owner;
 
+    @Schema(description = "List of asyncFetchedItems", implementation = Summary.class)
```
Comment (mkanal) [inline irs-models/src/main/java/net/catenax/irs/component/Job.java:110]:
```suggestion
    @Schema(description = "Summary of the job", implementation = Summary.class)
```

File: irs-models/src/main/java/net/catenax/irs/component/Job.java [hunk 2]
```diff
@@ -60,38 +62,58 @@ public class Job {
...
+    private Summary summary;
+
+    @Schema(description = "The given query parameters", implementation = QueryParameter.class)
```
Comment (mkanal) [inline irs-models/src/main/java/net/catenax/irs/component/Job.java:113]:
```suggestion
    @Schema(description = "The passed query parameters", implementation = QueryParameter.class)
```

**Review by mkanal — CHANGES_REQUESTED** [review]

File: irs-models/src/main/java/net/catenax/irs/component/Summary.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public class Summary {
      * asyncFetchedItems
      */
     @Schema(description = "Summary of the fetched jobs",
-            implementation = AsyncFetchedItems.AsyncFetchedItemsBuilder.class)
+            implementation = AsyncFetchedItems.class)
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply irs-models/src/main/java/net/catenax/irs/component/Summary.java:32]: Summary is the wrapper class, which contains the AsyncFetchedItems, so technically this is correct.
Do you want to move the content of AsyncFetchedItems to Summary?

File: irs-models/src/main/java/net/catenax/irs/component/ProtocolInformation.java [hunk 4]
```diff
@@ -24,14 +24,14 @@
...
 public class ProtocolInformation {
 
-    @Schema(description = "Uniform resource identifier of endpoint",
+    @Schema(description = "Uniform resource identifier of endpoint.",
             example = "https://catena-x.net/vehicle/basedetails/", implementation = java.net.URI.class)
     private String endpointAddress;
 
-    @Schema(description = "Protocol used to access the endpoint", example = "http or https",
-            implementation = java.net.URI.class)
+    @Schema(description = "Protocol used to access the endpoint.", example = "HTTP or HTTPS",
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply irs-models/src/main/java/net/catenax/irs/component/ProtocolInformation.java:31]: Why?

File: irs-models/src/main/java/net/catenax/irs/component/JobException.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+@Value
+@Builder(toBuilder = true)
+@JsonDeserialize(builder = JobException.JobExceptionBuilder.class)
+@AllArgsConstructor
+@SuppressWarnings("PMD.ShortClassName")
+@ExcludeFromCodeCoverageGeneratedReport
+public class JobException {
+
+    public static final int EXCEPTION_NAME_MAX_LENGTH = 100;
+    public static final int ZERO = 0;
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply irs-models/src/main/java/net/catenax/irs/component/JobException.java:34]: Removed it, why set min length to zero anyway?

**Review by mkanal — APPROVED** [review]

**Pull request merged**